lib/task-data/schemas/common-task-options.json

```diff
@@ -0,0 +1,30 @@
+{
+    "id": "rackhd/schemas/v1/tasks/common-options",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "RackHD Task Common Options",
+    "description": "The common options that shared by all tasks",
+    "type": "object",
+    "definitions": {
+        "Timeout": {
+            "type": "integer",
+            "minimum": -1,
+            "description": "This property can be used to specify a maximum amount of time a task may be run, in milliseconds. By default, this value is 24 hours. a value of 0 or -1 means infinite timeout"
+        },
+        "SchedulerOverrides": {
+            "type": "object",
+            "properties": {
+                "timeout": {
+                    "$ref": "#/definitions/Timeout"
+                }
+            }
+        }
+    },
+    "properties": {
```

Comment:
@yyscamper By convention, the definitions object encapsulates sub-schema that will be referenced elsewhere.  I think it would make more sense to move these properties into a schema called `options` within the `definitions` object. and change the id to `rackhd/schemas/v1/tasks/common`.  Other schema that reference this as a base would then use `$ref common/definitions/options`  This would allow us to easily add other common things in the future if needed.